gradle/runtime-jdk-provision.gradle

```diff
@@ -0,0 +1,25 @@
+import org.elasticsearch.gradle.OS
+import org.elasticsearch.gradle.VersionProperties
+
+apply plugin: 'elasticsearch.jdk-download'
+
+jdks {
+    provisioned_runtime {
+        vendor = VersionProperties.bundledJdkVendor
+        version = VersionProperties.bundledJdk
+        platform = OS.current().name().toLowerCase()
+    }
+}
+
+allprojects {
+    project.tasks.withType(Test).configureEach { Test test ->
+        test.dependsOn(rootProject.jdks.provisioned_runtime)
+        test.doFirst {
```

Comment:
`isRuntimeJavaHomeSet` is set during configuration time. There's no reason to defer this until execution by placing this in a `doFirst`.

Reply:
The `doFirst` is because we need to defer the download of the plugin.

Reply:
Yeah, but this kind of thing completely breaks input snapshotting. We cannot do this kind of thing because then we are effectively not tracking the runtime java version as an input.

We can accomplish the same thing by using a "lazy string" similar to what we do in the [`DistroTestPlugin`](https://github.com/elastic/elasticsearch/blob/12bb939e343225bdd7e9a8c7aff7f256e9455353/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java#L255-L263)